addons/account/tests/test_account_move_reconcile.py

```diff
@@ -5227,3 +5229,13 @@ def reconcile_move(move, transaction_amount, balance=None, date='2023-09-30'):
         self.assertEqual(payment1.state, 'paid')
         self.assertEqual(payment2.state, 'in_process')
         self.assertEqual(payment3.state, 'in_process')
+
+        customer_invoice_outstanding = self.init_invoice(move_type='out_invoice', amounts=[300], post=True)
+        payment1 = create_move_payment(customer_invoice_outstanding, 12, True)
+        payment2 = create_move_payment(customer_invoice_outstanding, 12)
+        reconcile_move(customer_invoice_outstanding, 12)
+        reconcile_move(customer_invoice_outstanding, 12)
+        self.assertEqual(payment1.state, 'in_process')
+        self.assertEqual(payment2.state, 'paid')
+        reconcile_move(payment1.move_id, 12)
```

Comment:
@william-andre Am I doing the payment reconciliation here correctly? The assertion in the following line fails because the state is `in_process` and I am not quite following what is going on :sweat_smile:

Reply:
The payments has one line on the receivable, and one line on the outstanding account.
The receivable account is getting matched with the invoice, so it cannot be reconciled anymore. (as the residual amount is expected to be 0; for full payments at least)
What you still need to reconcile at this point is the outstanding account, which typically comes from the statement line.

Once the outstanding are reconciled, the payment should go from "in process" to "paid"